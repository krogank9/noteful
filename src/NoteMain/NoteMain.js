import React from 'react';

import NoteItem from '../NoteItem/NoteItem.js';
import NotefulContext from "../NotefulContext.js";

class NoteMain extends React.Component {
	static contextType = NotefulContext;
	
	static defaultProps = {
		note: {}
	};
	
	render() {
		let noteId = this.props.match.params.noteId;
		let note = this.context.getNoteFromId(noteId);
		
		return (
			<div>
				<NoteItem note={note} />
				<p>{note.content}</p>
			</div>
		);
	}
}

export default NoteMain;
