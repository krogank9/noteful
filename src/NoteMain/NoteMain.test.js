import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NoteMain from './NoteMain'

describe(`NoteMain component`, () => {
	const props = {
		note: {
			id: 'a',
			name: 'test-class-name',
			modified: new Date(2018, 12, 15),
		}
	}

	it('renders a empty NoteMain by default', () => {
		const wrapper = shallow(<NoteMain />)
		expect(toJson(wrapper)).toMatchSnapshot()
	})

	it('renders NoteMain given props', () => {
		const wrapper = shallow(<NoteMain {...props} />)
		expect(toJson(wrapper)).toMatchSnapshot()
	})
})
