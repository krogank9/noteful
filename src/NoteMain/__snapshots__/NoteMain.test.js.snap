// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteMain component renders NoteMain given props 1`] = `
<div>
  <App
    note={
      Object {
        "id": "a",
        "modified": 2019-01-15T05:00:00.000Z,
        "name": "test-class-name",
      }
    }
  />
  <p />
</div>
`;

exports[`NoteMain component renders a empty NoteMain by default 1`] = `
<div>
  <App
    note={Object {}}
  />
  <p />
</div>
`;
